obok.py works under Cygwin command line e.g. $ python3 obok.py

natanb opened the pull request:
I try to use obok.py under Cygwin command line. After some troubles I managed to  use under Cygwin.
I made some corrections and add statements. If anyone is interested, tell me where I can upload the source. Thanks
natanb

Commit:
Added a code to remove Kindle watermark.
https://github.com/matimatik/DeDRM_tools/commit/f3fbc3573eadc4b505ed296176f35cd5a917cdbf

tracyde committed:
Support Python 2.7 and Python 3 winreg imports on Windows

journeyman88 committed:
Update obok.py

Changed MAC address fetching code to address possibile regression

Commit:
Issue 1635: Decypting PDF ebboks fixed

belm0 committed:
FAQ: note that Kindle 1.17 on Mac is 32-bit

ableeker committed:
Fix for decryption check

noDRM committed:
More PDF fixes

noDRM committed:
Remove broken CI

noDRM committed:
Add UUID to adobekey DER file names

noDRM committed:
Make keys fit into listbox

noDRM committed:
Update Readme

noDRM committed:
Improve key detection

noDRM committed:
Improve key detection for PDFs, too

noDRM committed:
Remove library book block

noDRM committed:
Remove library flag from MOBI book

noDRM committed:
Add old B&N algorihm (optional) just in case it's needed

noDRM committed:
Enable issue forms

noDRM committed:
Add IETF and Adobe font deobfuscation code

noDRM committed:
No longer break obfuscated fonts on DRM removal

noDRM committed:
Add useful error message for the new, uncracked ADEPT DRM

noDRM committed:
Add back Python2 support (ADEPT)

noDRM committed:
Fix ADE key import through plugin settings

noDRM committed:
Add B&N PDF DeDRM (untested), match UUID for Adobe PDFs

noDRM committed:
Remove CDP watermark from EPUBs

noDRM committed:
Auto-import keys from DeACSM plugin

noDRM committed:
Fix font deobfuscation for Python 2

noDRM committed:
Obok plugin cleanup

noDRM committed:
Add some more watermark removal code

noDRM committed:
Make CI auto-package the plugin

noDRM committed:
LCP support

noDRM committed:
Fixes a bug that sometimes caused the plugin to fail

Commit:
Fix typo in kindlekey.py that broke Mac version

Leseratte10 committed:
Fix username code for ADE key retrieval

noDRM committed:
Allow packaging without version number

noDRM committed:
Fix watermark stuff

noDRM committed:
Update to v10.0.2

noDRM committed:
Cleanup

noDRM committed:
Fix username decryption with unicode chars in Python2

noDRM committed:
Support importing multiple keys from ADE

noDRM committed:
Detect Kobo & Apple DRM in epubtest.py

noDRM committed:
Update changelog

noDRM committed:
Lots of B&N updates

noDRM committed:
Update PassHash documentation

noDRM committed:
Support for extracting PassHashes from ADE

noDRM committed:
Key retrieval updates

noDRM committed:
Edit .gitignore

noDRM committed:
Remove ancient code to import keys from ancient plugins

There were a couple specific DRM removal plugins before the DeDRM plugin
was created. These are obsolete since a long time, there's no need to
still have the code to import their config.

If people are still using these ancient plugins, they'll have to update
to an older version of DeDRM first, and then update to the current one.

noDRM committed:
Update watermark code

noDRM committed:
Ton of PDF DeDRM updates

- Support "Standard" and "Adobe.APS" encryptions
- Support decrypting with owner password instead of user password
- New function to return encryption filter name
- Support for V=5, R=5 and R=6 PDF files
- Support for AES256-encrypted PDF files
- Disable broken cross-reference streams in output

noDRM committed:
Fix broken key management

noDRM committed:
Update Changelog

noDRM committed:
Correct user pass padding, fix PDFStream export

ableeker committed:
Python 3 fixes

noDRM committed:
Cleanup

noDRM committed:
Begin work on standalone version

Now the plugin ZIP file (DeDRM_plugin.zip) can be run with a normal
Python interpreter as if it were a Python file (try
`python3 DeDRM_plugin.zip --help`). This way I can begin building a
standalone version (that can run without Calibre) without having to
duplicate a ton of code.

noDRM committed:
Bugfixes in standalone code for Calibre < 5 / Python 2

noDRM committed:
Add passhash interface to CLI

noDRM committed:
More work on standalone version, fix plugin

noDRM committed:
Try to fix B&N issues

noDRM committed:
Fix libcrypto DLL path search (see #13 and #14)

Co-authored-by: Adriano Caloiaro <code@adriano.fyi>

noDRM committed:
Fix FileNotFoundError during PassHash handling

noDRM committed:
Add "MemoryError" to FAQ

noDRM committed:
Make plugin work in Calibre 6 (Qt 6)

noDRM committed:
Fix IndexError in mobidedrm.py

noDRM committed:
Add empty placeholder file for LCP

noDRM committed:
Remove LCP references from Readme

noDRM committed:
Update FAQ

baby-bell committed:
Obok: Fix invalid UTF-8 causing UI to not open

For some reason, the title of a book on my device causes Obok to choke. Apparently it's not valid UTF-8.
This fixes that by ignoring decode errors.

a980e066a01 committed:
Remove OpenSSL support; only support PyCryptodome

This allows us to clean up the code a lot.

On Windows, it isn't installed by default and
most of the time not be found at all.

On M1 Macs, the kernel will kill the process instead.

Closes #33.

a980e066a01 committed:
Add support for "hardened" Adobe DRM

What took the most time was not reverse-engineering
the scheme, but actually finding books using it...

Closes #20, #25, #45

noDRM committed:
Update FAQs

Co-authored-by: ZolaLa <49111160+ZolaLa9@users.noreply.github.com>

noDRM committed:
Try to fix V3 PDF files

noDRM committed:
Debugging for __version issue

noDRM committed:
Make B&N plugin skip invalid hashes in Windows app

noDRM committed:
Improve error message

noDRM committed:
Update plugin readme

noDRM committed:
Add back unpad to fix Python2 support

noDRM committed:
(Hopefully) fix WineGetKeys for Kindle

noDRM committed:
Some untested Python2 Kindle bugfixes

noDRM committed:
Bugfix for Nook PDFs?

noDRM committed:
Fix print-replica Amazon books

noDRM committed:
Hopefully fix Kindle books

noDRM committed:
Fix Kindle for real

noDRM committed:
Fix Nook study key retrieval

sweetStreet committed:
fix the regular expression

noDRM committed:
Fix Obok plugin on Calibre 6 (#98)

noDRM committed:
Prepare release v10.0.3

noDRM committed:
Add some Python2 compat code I forgot to add earlier

noDRM committed:
Fix RSA.import_key (fixes #101)

Apparently "import_key" only exists in newer versions (as an alias to
"importKey"). "importKey" works in all versions ...

natanb commented:
[obok_br1.zip](https://github.com/apprenticeharper/DeDRM_tools/files/9169427/obok_br1.zip)
Here modify python source, that works under cygwin in command line:
$ python3 obok_br1.py
...

ElleKayEm commented:
As I said in the issue you started:

This repo is no longer maintained. Active work on DeDRM continues at noDRM's fork: https://github.com/noDRM/DeDRM_tools

noDRM committed:
Fix Calibre 6 issue in Obok plugin

noDRM committed:
Fix plugin crash with invalid ADE key

noDRM committed:
Fix ZIP attribute "external_attr" getting reset

noDRM committed:
Fix PDF corruption in Calibre 4 (#104)

noDRM committed:
Delete unused files

noDRM committed:
Another Python2 Bugfix for Obok

noDRM committed:
Move SafeUnbuffered to own Python file

noDRM committed:
Couple Python 2 fixes in (unsupported) standalone scripts

noDRM committed:
Remove AlfCrypto libraries and perform everything in Python

The old AlfCrypto DLL, SO and DYLIB files are ancient,
I don't have the systems to recompile them all, they
cause issues on ARM Macs, and I doubt with all the Python
improvements over the last years that they have a significant
performance advantage. And even if that's the case, nobody is
importing hundreds of DRM books at the same time so it shouldn't
hurt if some decryptions might take a bit longer.

noDRM committed:
Move unicode_argv to its own file

noDRM committed:
Cleanup

noDRM committed:
More stuff I missed

noDRM committed:
Cleanup / SafeUnbuffered bugfix

noDRM committed:
Fix DeACSM import for PDF files

noDRM committed:
Fix tons of PDF-related issues

RolandWH committed:
fix 2 spelling errors in FAQs.md

noDRM committed:
Fix key import sometimes generating corrupted keys.

Should fix #145, #134, #119, #116, #115, #109 and maybe others.

noDRM committed:
Cleanup

noDRM committed:
Update README, fixes #136

noDRM committed:
Update changelog

noDRM committed:
Add support for empty arrays (<>) in PDF objects. Fixes #183.

noDRM committed:
Strip whitespace from Kindle serials (#158)

noDRM committed:
Untested code for the Obok plugin to allow adding duplicate books.

See #148

noDRM committed:
Fix exception in error logging in ineptpdf

noDRM committed:
Fix Python3 bug in stylexml2css.py, fixes #232

noDRM committed:
PDF: Ignore invalid objid in non-strict mode, fixes #233

noDRM committed:
Update CHANGELOG

noDRM committed:
Try fixing a Python2 bug in the Obok plugin (#235)

noDRM committed:
Fix issue with remaining data in encryption.xml

noDRM committed:
Support for Adobe's 'aes128-cbc-uncompressed' encryption (see #242)

noDRM committed:
Try to add support for new K4PC

Co-authored-by: Andrew Innes <andrew.c12@gmail.com>
Co-authored-by: Satsuoni <satsuoni@hotmail.com>

noDRM committed:
Fix font decryption with multiple IDs (#347)

noDRM committed:
Fix python2 issues in Kindle and Nook code (#355)

noDRM committed:
Added fix for padding being correct on accident

Co-authored-by: Satsuoni <satsuoni@hotmail.com>

Satsuoni committed:
Added several more scramble functions to Kindle decrypt

noDRM committed:
Update CI to create an automatic beta release

noDRM committed:
Disable auto-prerelease again (#358)

noDRM committed:
Add warning to the standalone code

noDRM committed:
Prepare release candidate v10.0.9

noDRM committed:
Fix PDF decryption of ancient 40-bit RC4 with R=2

noDRM committed:
Fix PDF decryption for 256-bit AES with V=5

noDRM committed:
Fix python2 issues in kgenpids and kindlekey

noDRM committed:
More Python2 bugfixes

noDRM committed:
More Py2 fixes

noDRM committed:
More Python2 fixes

noDRM committed:
Enable autorelease into 2nd repo

noDRM committed:
Fix more Py2/Py3 stuff

precondition committed:
Catch FileNotFoundError due to undownloaded ebooks

precondition committed:
Obok.py/action.py: invoke _() only once

noDRM committed:
Fix file lock issue in androidkindlekey.py

noDRM committed:
Bunch of updates for the FileOpen script

noDRM committed:
More FileOpen fixes

noDRM committed:
Update kfxdedrm as suggested in #440

noDRM committed:
Add binascii import, fixes FileOpen #514

unwiredben committed:
Fix crash using bare sha1 symbol

Use sha1 from hashlib, as it isn't imported globally, fixed crash trying to decrypt a eReader PDB file

C0rn3j committed:
DeDRM ion: Clean out errorneous whitespace and UTF8 definition from python 2 times

C0rn3j committed:
DeDRM ion: Correctly throw last exception if decrypt fails

jcotton42 committed:
Fix Obok import in Calibre flatpak by using /sys/class/net/IFACE/address instead of `ip` (#586)

Fix #585.
Use /sys/class/net/IFACE/address for the MAC address instead of the ip
command.

noDRM committed:
Fix CI

noDRM committed:
Update changelog

noDRM committed:
Update obok readme to include wmic requirement (fixes #670)

noDRM committed:
Add note about key import/export for K4PC in the help file (fixes #663)

noDRM committed:
Don't repack EPUB if nothing has changed

noDRM committed:
Fix python2 support for ion.py

noDRM committed:
Remove future import from ion.py
